subprojects/core-api/src/main/java/org/gradle/tooling/provider/model/ParameterizedToolingModelBuilder.java

```diff
@@ -32,13 +35,49 @@
  * </p>
  *
  * @param <T> The type of parameter used by this model builder.
- * @since 4.4
  * @see ToolingModelBuilder
+ * @since 4.4
  */
 public interface ParameterizedToolingModelBuilder<T> extends ToolingModelBuilder {
+
     /**
      * Returns the expected type of the parameter.
-     * It should be an interface with only getters and setters.
+     * <p>
+     * <b>Contract</b>
+     * <p>
+     * The parameter type must conform to the following rules:
+     * <ul>
+     *     <li>It must be an interface.</li>
+     *     <li>It must contain only getter-like and setter-like methods.</li>
```

Comment:
I'm not sure this one is true. I think the interface can have whatever methods the author likes.

Reply:
Sorry, it looks like it is the case at the moment. But it shouldn't be, the restrictions don't make any sense. The parameters should have the same restrictions as other user types passed via the tooling API.

Reply:
I was also surprised by the restrictions. It would be great to relax them in the future 👍